R/dosing_nomograms.R

```diff
@@ -0,0 +1,23 @@
+#' Weight-based starting dose (e.g., mg/kg)
+#'
+#' @inheritParams model_based_starting_dose
+#' @param args something
+#' @export
+
+weight_based_starting_dose <- function(covariates, design, cov_mapping) {
+  reg_def <- design$initial_regimen$regimen
+
+  if (is.null(design$initial_regimen$mgkg)) {
+    stop(
+    "For weight-based initial dosing, ",
+    "specify `create_initial_regimen_design(mgkg = ...)`"
+    )
+  }
+  if (is.null(covariates$WT$value)) {
```

Comment:
should we use covariate mapping here? The user does create a covariate map. Not sure if it makes it easier / more complex.

Reply:
we could make it customizable which covariate is used for weight, but it would have to be specified in the `create_initial_regimen_design` call and not in cov_mapping, since cov_mapping is model-to-data mapping and does not necessary contain info about weight.

we could imagining something like this:

```r
  init_reg <- create_initial_regimen_design(
    method = weight_based_starting_dose,
    mgkg = 15,
    weight_cov = "WT",
    regimen = list(
      interval = 12,
      t_inf = 1.5,
      n = 10,
      type = "infusion"
    )
  )
```

This would allow the user to also do the following to calculate it based on something like IBW:
```r
  init_reg <- create_initial_regimen_design(
    method = weight_based_starting_dose,
    mgkg = 15,
    weight_cov = "IBW",
    regimen = list(
      interval = 12,
      t_inf = 1.5,
      n = 10,
      type = "infusion"
    )
  )
```

Reply:
Yeah like this, more generic. But perhaps `weight_cov` doesn't need to be an argument to `create_initial_regimen_design`, can we just depend on passing via `...`? OTOH weight is often useful for initial dose, so I don't mind adding it as an argument.

Reply:
@roninsightrx I have updated this function to be more generic, it will be passed to `create_initial_regimen_design` with `...` args, but it will be required if this method is called, for obvious reasons.